redfish-core/lib/memory.hpp

```diff
@@ -846,7 +1009,31 @@ inline void requestRoutesMemory(App& app)
             return;
         }
 
-        getDimmData(asyncResp, dimmId);
+        asyncResp->res.jsonValue["@odata.type"] = "#Memory.v1_12_0.Memory";
+        asyncResp->res.jsonValue["@odata.id"] =
+            boost::urls::format("/redfish/v1/Systems/system/Memory/{}", dimmId);
+
+        getDimmObject(asyncResp, dimmId, getDimmData);
+    });
```

Comment:
I think we can refactor the body L989-L1017 into a function (e.g. `handleDimmObjectGet()`).

`getDimmObject()` is used only and it is intended to call `getDimmData()` (via template function `handler()`).
I think we can remove the template `handler()` but instead calling `getDimmData()` inside `handleDimmObject()`.

```
handleDimmObject()
{
       ...
        handler(asyncResp, dimmId, objectPath, serviceMap);
        -->
         getDimmData(asyncResp, dimmId, objectPath, serviceMap);
}
```

I think this would simplify the code flow better.


I also think we can move `getDimmObject()` logic inside  `handleDimmObjectGet()` [after refactor] like
```
handleDimmObjectGet()
{
       ...
    // GetSubTree on all interfaces which provide info about a Memory
    dbus::utility::getSubTree(
        "/xyz/openbmc_project/inventory", 0, memInterfaces,
       std::bind_front( handleDimmObject, ...);
}
```

Reply:
I would like to defer this change until the rebase of the other related commit. (https://github.com/ibm-openbmc/bmcweb/pull/571) That one gets rid of the <template> stuff and restructures much like you suggest here. I think it will make it easier to do the rebase if this one is closer to the original change so there will be less merge conflicts on the rebase.

Please let me know if you are okay with that approach.

Reply:
I'm good.